src/dfx-core/src/config/model/local_server_descriptor.rs

```diff
@@ -173,10 +181,14 @@ impl LocalServerDescriptor {
         path.exists().then(|| load_json_file(&path)).transpose()
     }
 
-    pub fn settings_digest(&self) -> &str {
-        self.settings_digest
-            .as_ref()
-            .expect("settings_digest must be set")
+    pub fn settings_digest(&self) -> String {
+        self.settings_digest.clone().unwrap_or_else(|| {
+            let network_id_contents = crate::fs::read(&self.data_directory.join("network-id"))
+                .expect("Could not read network-id file.");
+            let network_metadata: NetworkMetadata = serde_json::from_slice(&network_id_contents)
+                .expect("Could not parse network-id contents.");
+            network_metadata.settings_digest
```

Comment:
Please move this to a separate load method that returns a `Result`. This method panics if the settings digest isn't set because it should only be called if the settings digest was set earlier.

Reply:
OK. Done.